### ISSUE TITLE
Testing the "Resizable ArrayBuffer" proposal

### ISSUE THREAD

**Issue by jugglinmike** [tc39/test262#3003]
https://github.com/tc39/proposal-resizablearraybuffer

This proposal advanced to stage 3 on 2021-05-25

**Comment by jugglinmike** [comment]
I've begun writing these tests

**Comment by jugglinmike** [comment]
I've submitted these tests for review, split up across 11 pull requests in order to facilitate incremental review (the pull requests are listed above).

There's one design decision which effected multiple patches, so to keep things organized, I'll describe it here:

Hosts are permitted to fail any resize operation at their own discretion. This is fairly straightforward to accommodate in direct tests (e.g. those for `ArrayBuffer.prototype.resize`), but it's more difficult to address for the features which integrate.

The approach I've settled on involves dynamically assigning assertion expectation values within `try`/`catch` statements. This is most jarring in those tests which concern abrupt completions: they use `Test262Error` and `assert.throws` to verify that a given invocation does *not* throw. In all cases, the design is intended to maximize coverage (by validating *something* for the "host-defined failure" case) while limiting the amount of branching in the test body (which is generally an invitation for test bugs and reader confusion).

This is all a little unorthodox, but I feel it's warranted by the proposal's unique behavior.

/cc @syg
